### PR TITLE
Modal doc improvements

### LINKED ISSUE #2120: Differentiate between Modal Props and ModalSetting attributes in props tab
### Link to the Page

https://www.skeleton.dev/utilities/modals

### Describe the Issue (screenshots encouraged!)

People keep trying to mix them interchangeably because they are part of the same list.
We should consider ways to split them without breaking out of the docs shell to much.

I volunteer, mostly because it's been bugging me for a while now :)

### PR DESCRIPTION
**Pull request by endigo9740**
## Linked Issue

Closes #2120

## Description

Makes a number of improvements to the modal documentation:

1. Provides a better explanation of Modal Settings vs Modal Props
2. Major reworks to the Component Modals section
3. Component Modal examples now use the standard Previewer format
4. The Register/Direct methods are no longer hidden in the Previewer code view
5. Greatly simplified the examples
6. Rewrote most accordion sections; each include proper code snippets

Overall this should be MUCH more accessible.

## Changsets

Instructions: Changesets automate our changelog. If you modify files in `/packages/skeleton`, run `pnpm changeset` in the root of the monorepo, follow the prompts, then commit the markdown file. Changes that add features should be `minor` while chores and bugfixes should be `patch`. Please prefix the changeset message with `feat:`, `bugfix:` or `chore:`.

## Checklist

Please read and apply all [contribution requirements](https://www.skeleton.dev/docs/contributing).

- [x] This PR targets the `dev` branch (NEVER `master`)
- [x] Documentation reflects all relevant changes
- [x] Branch is prefixed with: `docs/`, `feat/`, `chore/`, `bugfix/`
- [x] Ensure Svelte and Typescript linting is current - run `pnpm check`
- [x] Ensure Prettier linting is current - run `pnpm format`
- [x] All test cases are passing - run `pnpm test`
- [x] Includes a changeset (if relevant; see above)

### CODE REVIEW

**Comment by endigo9740** [comment]
@Sarenor I'll add you as the reviewer on this since you suggested the original ticket. Feel free to double check spelling and grammar please!

File: sites/skeleton.dev/src/routes/(inner)/docs/themes/+page.svelte [hunk 1]
```diff
@@ -213,7 +213,7 @@ body { background: red; }
 				</p>
 			</div>
 			<div>
-				<a class="btn variant-filled-secondary font-bold" href="https://csshero.org/mesher/" target="_blank" rel="noreferrer">
+				<a class="btn variant-filled font-bold" href="https://csshero.org/mesher/" target="_blank" rel="noreferrer">
```
Comment (Sarenor) [inline sites/skeleton.dev/src/routes/(inner)/docs/themes/+page.svelte:216]: Was this change intentional or did it slip in from some previous experiments?

File: sites/skeleton.dev/src/routes/(inner)/utilities/modals/+page.svelte [hunk 2]
```diff
@@ -325,7 +350,8 @@ new Promise<boolean>((resolve) => {
 				<h2 class="h2">Component Modals</h2>
 				<span class="badge variant-filled-warning">Advanced</span>
 			</div>
-			<p>To create custom modals, generate a new component, then pass this to the Modal system.</p>
+			<p>Skeleton allows you to generate custom modals using Svelte components. See several examples provided below.</p>
```
Comment (Sarenor) [inline sites/skeleton.dev/src/routes/(inner)/utilities/modals/+page.svelte:353]: > See several examples provided below.

I'm not sure if I like this sentence, but

> Check out the examples provided below

isn't better.

**Review by Sarenor — APPROVED** [review]
Overall, no complaints or typos found.
Thanks for handling this, that was a more sweeping but better change than I had in mind.

On the other hand a more contained one, I was thinking about expanding the DocsShell to allow for some form of arbitrary content in the props tab. Definitely like this way better, let's hope it's obvious enough!

File: sites/skeleton.dev/src/routes/(inner)/docs/themes/+page.svelte (hunk 1 above)
Comment (endigo9740) [reply sites/skeleton.dev/src/routes/(inner)/docs/themes/+page.svelte:216]: Intended. I copied the style and fixed it as I went.

File: sites/skeleton.dev/src/routes/(inner)/utilities/modals/+page.svelte (hunk 2 above)
Comment (endigo9740) [reply sites/skeleton.dev/src/routes/(inner)/utilities/modals/+page.svelte]: During the ViteConf talk they had a great segment on docs - one of the key takeaways is less is more. I'll nix it. The examples should be visible and obvious without written direction.

**Comment by endigo9740** [comment]
@Sarenor

> On the other hand a more contained one, I was thinking about expanding the DocsShell to allow for some form of arbitrary content in the props tab. Definitely like this way better, let's hope it's obvious enough!

I'm foreseeing some wider reaching changes to the Doc layouts in the future. Keep that in mind and let's discuss then.
